src/main/java/org/peergos/Server.java

```diff
@@ -13,106 +15,209 @@
-                throw new IllegalStateException("Unable to determine API port");
+    public Server() throws Exception {
+        String home = System.getenv("HOME");
+        Path configPath = Path.of(home, ".ipfs");
+
+        Logging.init(configPath);
+        Config config = readConfig(configPath);
+        System.out.println("Starting Nabu version: " + APIService.CURRENT_VERSION);
+        String specType = config.getProperty("Datastore", "Spec", "type");
+        if (!specType.equals("mount")) {
+            throw new IllegalStateException("Unable to read mount configuration");
+        }
+        List<Map<String, Object>> mounts = config.getPropertyList("Datastore", "Spec", "mounts");
+        Optional<Map<String, Object>> blockMountOpt = mounts.stream().filter(m -> m.get("mountpoint").equals("/blocks")).findFirst();
+        if (blockMountOpt.isEmpty()) {
+            throw new IllegalStateException("Unable to fine /blocks mount");
+        }
+        Config blockConfig = new Config(blockMountOpt.get());
+        String blockPath = blockConfig.getProperty("child", "path");
+        String blockShardFunc = blockConfig.getProperty("child", "shardFunc");
+        String blockType = blockConfig.getProperty("child", "type");
+        if (!(blockPath.equals("blocks") && blockShardFunc.equals("/repo/flatfs/shard/v1/next-to-last/2")
+            && blockType.equals("flatfs"))) {
+            throw new IllegalStateException("Expecting flatfs mount at /blocks");
+        }
+        Path blocksPath = Path.of(System.getenv("HOME"), ".ipfs", "blocks");
+        File blocksDirectory = blocksPath.toFile();
+        if (!blocksDirectory.exists()) {
+            if (!blocksDirectory.mkdir()) {
+                throw new IllegalStateException("Unable to make blocks directory");
             }
-            int apiPort = Integer.parseInt(apiPortStr); // 5001
-            InetSocketAddress localAPIAddress = new InetSocketAddress("localhost", apiPort);
-
-            int maxConnectionQueue = 500;
-            int handlerThreads = 50;
-            HttpServer apiServer = localAPI.initAndStart(localAPIAddress, node, blocks, maxConnectionQueue, handlerThreads);
-
-            Thread shutdownHook = new Thread(() -> {
-                System.out.println("Stopping server...");
-                try {
-                    node.stop().get();
-                    apiServer.stop(3); //wait max 3 seconds
-                } catch (InterruptedException | ExecutionException ex) {
-                    ex.printStackTrace();
-                }
-            });
-            Runtime.getRuntime().addShutdownHook(shutdownHook);
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+        } else if(blocksDirectory.isFile()) {
+            throw new IllegalStateException("Unable to create blocks directory");
         }
+        FileBlockstore blocks = new FileBlockstore(blocksPath);
+
+        int hostPort = 6001; // 6001
```

Comment:
we should get this from the config

Reply:
i was wondering which line in the config it referes to. Can you point me to it? No occurence of 6001 afaict

Reply:
i extract the Addresses -> API MultiAddress for use by the java-ipfs-http-client RPC client.